gcloud/bigquery/dataset.py

```diff
@@ -0,0 +1,339 @@
+    @default_table_expiration_ms.setter
+    def default_table_expiration_ms(self, value):
+        """Update default expiration time for tables in the dataset.
+
+        :type value: ``int``, or ``NoneType``
+        :param value: new default time, in milliseconds
+        """
+        if not isinstance(value, int) and value is not None:
+            raise ValueError()
+        self._properties['defaultTableExpirationMs'] = value
+
+    @property
+    def description(self):
+        """Description of the dataset.
+
+        :rtype: ``str``, or ``NoneType``
+        """
+        return self._properties.get('description')
+
+    @description.setter
+    def description(self, value):
+        """Update description of the dataset.
+
+        :type value: ``str``, or ``NoneType``
+        :param value: new description
+        """
+        if not isinstance(value, str) and value is not None:
+            raise ValueError()
+        self._properties['description'] = value
+
+    @property
+    def friendly_name(self):
+        """Title of the dataset.
+
+        :rtype: ``str``, or ``NoneType``
+        """
+        return self._properties.get('friendlyName')
+
+    @friendly_name.setter
+    def friendly_name(self, value):
+        """Update title of the dataset.
+
+        :type value: ``str``, or ``NoneType``
+        :param value: new title
+        """
+        if not isinstance(value, str) and value is not None:
+            raise ValueError()
+        self._properties['friendlyName'] = value
+
+    @property
+    def location(self):
+        """Location in which the dataset is hosted.
+
+        :rtype: ``str``, or ``NoneType``
+        """
+        return self._properties.get('location')
+
+    @location.setter
+    def location(self, value):
+        """Update location in which the dataset is hosted.
+
+        :type value: ``str``, or ``NoneType``
+        :param value: new location
+        """
+        if not isinstance(value, str) and value is not None:
+            raise ValueError()
+        self._properties['location'] = value
+
+    def _require_client(self, client):
+        """Check client or verify over-ride.
+
+        :type client: :class:`gcloud.bigquery.client.Client` or ``NoneType``
+        :param client: the client to use.  If not passed, falls back to the
+                       ``client`` stored on the current dataset.
+
+        :rtype: :class:`gcloud.bigquery.client.Client`
+        :returns: The client passed in or the currently bound client.
+        """
+        if client is None:
+            client = self._client
+        return client
+
+    def _set_properties(self, api_response):
+        """Update properties from resource in body of ``api_response``
+
+        :type api_response: httplib2.Response
+        :param api_response: response returned from an API call
+        """
+        self._properties.clear()
+        cleaned = api_response.copy()
+        cleaned['creationTime'] = float(cleaned['creationTime'])
```

Comment:
The back-end passes it as a string.  We want to convert it to a `datetime` instance (both `created` and `modified` are read-only, so we don't care about the other direction).